### ISSUE TITLE
Provide binary releases?

### ISSUE THREAD

**Issue by nilsbyte** [kcat/dsoal#13]
Hello @kcat, can you maybe provide a binary for each release? It's very cumbersome to compile this from source for somebody that isn't familiar with this.

**Comment by kcat** [comment]
I just added basic CPack support to the project, so what I'm going to try to do is set up AppVeyor to build the project and produce a binary package as a build artifact. I haven't really used CPack before and I'm not really sure how to get AppVeyor to generate the package that can be hosted/downloaded, but I'm looking into it.

**Comment by ThreeDeeJay** [comment]
> 
> 
> Hello @kcat, can you maybe provide a binary for each release? It's very cumbersome to compile this from source for somebody that isn't familiar with this.

Latest binaries available here: https://ci.appveyor.com/project/ChrisRobinson/dsoal
Select Win32/Win64 job > Artifacts > dsoal.zip (it says it expires, so grab it while supply last lol)
Alternatively, there's this, tho I'm not sure how trustworthy the source is: http://vaporeon.io/hosted/dsoal-builds/

**Comment by SkyNinja** [comment]
> > Hello @kcat, can you maybe provide a binary for each release? It's very cumbersome to compile this from source for somebody that isn't familiar with this.
> 
> Latest binaries available here: https://ci.appveyor.com/project/ChrisRobinson/dsoal
> Select Win32/Win64 job > Artifacts > dsoal.zip (it says it expires, so grab it while supply last lol)
> Alternatively, there's this, tho I'm not sure how trustworthy the source is: http://vaporeon.io/hosted/dsoal-builds/

Appveyor deletes artifacts over a month old and whoever was hosting on that other site seems to be MIA, unfortunately.

**Comment by ThreeDeeJay** [comment]
> 
> 
> > Latest binaries available here: https://ci.appveyor.com/project/ChrisRobinson/dsoal
> > Select Win32/Win64 job > Artifacts > dsoal.zip (it says it expires, so grab it while supply last lol)
> > Alternatively, there's this, tho I'm not sure how trustworthy the source is: http://vaporeon.io/hosted/dsoal-builds/
> 
> Appveyor deletes artifacts over a month old and whoever was hosting on that other site seems to be MIA, unfortunately.

@SkyNinja Unfortunate. Guess the only alternative to get official builds that expired would be forking DSOAL, making an Appveyor account and building it on your own, it's fairly easy. [This guide](https://stefanscherer.github.io/fork-appveyor-buildpipeline/) helped me set it up.

**Comment by ThreeDeeJay** [comment]
> [CrezyDud](https://github.com/CrezyDud): just build it yourself

> [nilsbyte](https://github.com/nilsbyte) (OP): It's very cumbersome to compile this from source for somebody that isn't familiar with this.

@CrezyDud Here's the point you missed lol

> why setup appveyor?

@CrezyDud Because _back then_ not even the GitHub Actions PR existed, let alone merged, so the only official (and trustworthy) binaries were on AppVeyor, which had already expired, so at least a forked AppVeyor _was_ a viable alternative.

**Comment by ThreeDeeJay** [comment]
Anyway, I think it would be nice if someone (@Deledrius *wink-wink) could set it up to automatically push GitHub Actions artifacts as releases, unless that's reserved for stable builds.
That would allow directly linking to the latest, non-expiring official binaries and download wouldn't need an explanation of how to navigate to the download, or at least we could add such explanation to the README 🤔

**Comment by Deledrius** [comment]
> Anyway, I think it would be nice if someone (@Deledrius *wink-wink) could set it up to automatically push GitHub Actions artifacts as releases

I'll see what I can do.  I wasn't sure if @kcat has a particular preference on how those would be triggered/tagged.

**Comment by ThreeDeeJay** [comment]
> > Anyway, I think it would be nice if someone (@Deledrius *wink-wink) could set it up to automatically push GitHub Actions artifacts as releases
> 
> I'll see what I can do. I wasn't sure if @kcat has a particular preference on how those should be triggered/tagged.

Just occurred to me that they could be marked as Pre-release so it's clear they're not explicitly stable but still show up in the Releases section when clicking on it. This way, artifacts wouldn't just expire like Github Actions and AppVeyor.
As for versioning, that's a good point. Perhaps they could just use the commit short hash as tag or just the date.
But hey, that's just my 2 cents so I'm also curious about @kcat's stance 👀

**Comment by kcat** [comment]
The main issue I have is that I'm not sure where to tag (pre-)releases since the codebase is being continually developed. There aren't any milestone goals, there's known bugs that can't be fixed any time soon (panning stuff mostly), so except for the rare cases of fixes or features that require multiple commits made over multiple days, each round of commits would essentially be its own release candidate.

There's also the issue that it needs to be packaged with recent Git builds of OpenAL Soft. Unless and until it can be built-in/static linked into DSOAL, you need to include `dsoal-aldrv.dll` with it for it to work.

I'm certainly not against having packaged "releases" of some sort, even pulled automatically from artifacts (as long as they're clearly marked with the Git commit hash they're built from). Just need to make sure it has all the necessary pieces to be usable.

**Comment by Deledrius** [comment]
@kcat That's good to know.  I think it should be possible to meet those constraints.

**Comment by ThreeDeeJay** [comment]
Yeah, packaging with the latest OpenAL Soft (especially with the right architecture) does sound really tricky.
Perhaps adding OpenAL Soft as a submodule (folder shortcut to the OALS repo) could work by building OALS every time.
Anyhow, I came across a really interesting doc with a template to push Appveyor artifacts as Github (pre-)releases, even allows using the Appveyor build number as a tag/version which I'm not sure can be done via Github Actions https://www.appveyor.com/docs/deployment/github/

**Comment by mirh** [comment]
Just for the records, openal-soft is missing that juicy artifacts button inside actions.

**Comment by ThreeDeeJay** [comment]
> Just for the records, openal-soft is missing that juicy artifacts button inside actions.

@mirh Just submitted a [PR](https://github.com/kcat/openal-soft/pull/708) for that.

**Comment by ThreeDeeJay** [comment]
@nilsbyte It's your lucky day 👀 https://github.com/kcat/dsoal/releases 
There's older builds here too, in case they're needed:
https://github.com/ThreeDeeJay/dsoal/releases
http://vaporeon.io/hosted/dsoal-builds/old/

**Comment by nilsbyte** [comment]
@ThreeDeeJay lol completely forgot about that. It’s been almost exactly 5 years since I opened this issue. Thank you for the effort in providing compiled versions. That’s really much appreciated. Without projects like this, retro gaming would be not be possible at these high standards.
